pom.xml

```diff
@@ -172,6 +172,31 @@ under the License.
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-compiler-plugin</artifactId>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-checkstyle-plugin</artifactId>
+                <version>3.0.0</version>
+                <dependencies>
+                    <dependency>
+                        <groupId>com.puppycrawl.tools</groupId>
+                        <artifactId>checkstyle</artifactId>
```

Comment:
The license of checkstyle is [LGPL 2.1](https://github.com/checkstyle/checkstyle/blob/master/LICENSE), it is in [CATEGORY X](https://www.apache.org/legal/resolved.html#what-can-we-not-include-in-an-asf-project-category-x), which could not be included in our release for opensource.
Is it necessary for this PR? if not, I recommend you remove it first.

Reply:
Yes,it's just used when I check the code myself.